go/oasis-test-runner/scenario/e2e/genesis_file.go

```diff
@@ -93,15 +98,45 @@ func (s *genesisFileImpl) Run(childEnv *env.Env) error {
 	}
 	s.Logger.Info("genesis file from dumped network state passed genesis check command")
 
-	uncanonicalPath := filepath.Join(childEnv.Dir(), "genesis_uncanonical.json")
-	if err = s.createUncanonicalGenesisFile(childEnv, uncanonicalPath); err != nil {
-		return fmt.Errorf("e2e/genesis-file: creating uncanonical genesis file failed: %w", err)
+	// Check if the latest Mainnet genesis file passes genesis check.
+	latestMainnetGenesis := filepath.Join(childEnv.Dir(), "genesis_mainnet.json")
+
+	if err = s.downloadLatestMainnetGenesisFile(childEnv, latestMainnetGenesis); err != nil {
+		return fmt.Errorf("e2e/genesis-file: failed to download latest Mainnet genesis "+
+			"file at '%s': %w", latestMainnetGenesisURL, err)
 	}
-	err = s.runGenesisCheckCmd(childEnv, uncanonicalPath)
+	err = s.runGenesisCheckCmd(childEnv, latestMainnetGenesis)
 	expectedError := "genesis file is not in canonical form, see the diff on stderr"
 	switch {
 	case err == nil:
-		return fmt.Errorf("e2e/genesis-file: running genesis check for an uncanonical genesis file should fail")
+		return fmt.Errorf("e2e/genesis-file: running genesis check for the latest Mainnet"+
+			" genesis file at '%s' should fail with '%s'",
+			latestMainnetGenesisURL, expectedError,
+		)
+	case !strings.Contains(err.Error(), expectedError):
+		return fmt.Errorf(
+			"e2e/genesis-file: running genesis check for the latest Mainnet genesis "+
+				"file should fail with an error containing: '%s' (actual error: %s)",
+			expectedError, err,
+		)
+	default:
+		s.Logger.Info("latest Mainnet genesis file is OK, but is not in canonical form")
+	}
+
```

Comment:
The test should probably also make sure that the genesis document hash matches?

Reply:
Yes, good idea.

Reply:
Done.